### PR TITLE
move eager connet to userProvider

### PR DESCRIPTION
**Pull request by MohammadPCh**
- #542

### CODE REVIEW

File: src/context/UserProvider.tsx [hunk 1]
```diff
@@ -51,15 +51,24 @@ const UserContext = createContext<IUserContext>({
...
 
+	useEffect(() => {
+		const selectedWalletName = localStorage.getItem(StorageLabel.WALLET);
+		const wallet = walletsArray.find(w => w.value === selectedWalletName);
+		if (wallet) {
+			activate(wallet.connector, err => {
+				console.log('err', err);
+			});
+		}
+	}, [activate]);
```
Comment (RamRamez) [inline src/context/UserProvider.tsx:70]: `activate` is a always a function and I think adding it as a dependency is not necessary. If user locks his wallet and unlock again, the app won't try to `activate`, So I think you should replace `activate` dependency with web3React `active`

**Review by RamRamez — CHANGES_REQUESTED** [review]

File: src/context/UserProvider.tsx [hunk 2]
```diff
@@ -13,9 +13,9 @@ import { initializeApollo } from '@/apollo/apolloClient';
 import { GET_USER_BY_ADDRESS } from '@/apollo/gql/gqlUser';
 import { compareAddresses, showToastError, signMessage } from '@/lib/helpers';
 import { getToken } from '@/services/token';
-import useWallet from '@/hooks/walletHooks';
```
Comment (RamRamez) [inline src/context/UserProvider.tsx:16]: this file is not used anymore, so I think you can remove it completely.

**Review by RamRamez — CHANGES_REQUESTED** [review]

File: src/context/UserProvider.tsx (hunk 2 above)
Comment (MohammadPCh) [reply src/context/UserProvider.tsx:16]: exactly ;)
Thank you @RamRamez

File: src/context/UserProvider.tsx (hunk 1 above)
Comment (MohammadPCh) [reply src/context/UserProvider.tsx:70]: I put this for the time that the activate function changed, for example, if it is not initial yet.

**Review by RamRamez — APPROVED** [review]

**Pull request merged**